src/test/java/BaseTest.java

```diff
@@ -1,11 +1,104 @@
+    String playListName = "TestPro Playlist";
+    String songName = "grav";
+    String userName =  "demo@testpro.io";
+    String userPassword = "te$t$tudent";
+
+    public static WebDriver driver = null;
+    public static ChromeOptions options = new ChromeOptions();
+    public static WebDriverWait wait;
+    public static Wait<WebDriver> fluentWait;
+    public static Actions actions = null;
+
     @BeforeSuite
     static void setupClass() {
         WebDriverManager.chromedriver().setup();
     }
+
+    @BeforeMethod
+    @Parameters({"BaseURL"})
+    public void launchBrowser(String BaseURL) throws InterruptedException {
+
+    //  Pre-condition
+    //  Added ChromeOptions argument below to fix websocket error
+        ChromeOptions options = new ChromeOptions();
+        options.addArguments("--remote-allow-origins=*","--disable-notifications","--incognito");
+        options.setExperimentalOption("excludeSwitches", new String[]{"enable-automation"});
+
+        driver = new ChromeDriver(options);
+        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
+        wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+        actions = new Actions(driver);
+        fluentWait = new FluentWait<WebDriver>(driver)
+                .withTimeout(Duration.ofSeconds(5))
+                .pollingEvery(Duration.ofMillis(200))
+                .ignoring(TimeoutException.class);
+
+        url = BaseURL;
+        navigateToPage();
+    //  enterEmail(userName);
+    //  enterPassword(userPassword);
+    //  submit();
+    //  Thread.sleep(3000);
+
+    //  wait = new WebDriverWait();
+    //  driver.manage().window().maximize();
+    }
+
+    @AfterMethod
+    public void closeBrowser(){
+    //  driver.quit();
+    }
+
+    protected void submit() {
+    //  WebElement submit = driver.findElement(By.cssSelector("button[type='submit']"));
+        WebElement submit = wait.until
+                (ExpectedConditions.visibilityOfElementLocated
+                        (By.cssSelector("button[type='submit']")));
+        submit.click();
+    }
+
+    protected void enterPassword(String password) {
+    //  WebElement passwordField = driver.findElement(By.cssSelector("input[type='password']"));
+        WebElement passwordField = wait.until
+                (ExpectedConditions.visibilityOfElementLocated
+                        (By.cssSelector("input[type='password']")));
+        passwordField.clear();
+        passwordField.sendKeys(password);
+    }
+
+    protected void enterEmail(String email) {
+    //  WebElement emailField = driver.findElement(By.cssSelector("input[type='email']"));
+        WebElement emailField = wait.until
+                (ExpectedConditions.visibilityOfElementLocated
+                        (By.cssSelector("input[type='email']")));
+        emailField.clear();
+        emailField.sendKeys(email);
+    }
+
+    protected void navigateToPage() {
+        driver.get(url);
+    }
```

Comment:
Page Object Model (POM): recommended to associate the methods with the corresponding page class where they are relevant. In the POM, each page or section of a web application is represented by a separate class. These classes encapsulate the elements and actions specific to that page or section.

Move these methods to the corresponding class according to the POM.

For example
![Screenshot_2](https://github.com/YuriiLyndiuk/A57-QA-Automation-Framework/assets/104761482/b07cd7c1-7977-4e53-80ea-8045ac17f779)